web-common/src/features/canvas/components/charts/types.ts

```diff
@@ -41,8 +41,15 @@ export type ChartDataResult = {
   isFetching: boolean;
   fields: ChartFieldsMap;
   error?: HTTPError | null;
+  metadata?: ChartMetadata;
 };
 
+export interface ChartMetadata {
+  xValues?: string[];
+  colorValues?: string[];
+  yValues?: string[];
+}
```

Comment:
Nit: I think this seems more like "config" than "metadata"? For example, I don't think of "metadata" as being something that the developer can "reset".